### PR TITLE
fix: implement setMetadata in HmacKey and fix associated tests

### PR DESCRIPTION
**Pull request by ddelgrosso1**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: conformance-test/libraryMethods.ts [hunk 1]
```diff
@@ -680,8 +680,8 @@ export async function deleteHMAC(options: ConformanceTestOptions) {
   const metadata = {
     state: 'INACTIVE',
   };
-  options.hmacKey!.setMetadata(metadata);
-  await options.hmacKey!.delete();
+  await hmacKey.setMetadata(metadata);
```
Comment (shaffeeullah) [inline conformance-test/libraryMethods.ts:683]: does this not run into the problem where the instructions get used here?

File: src/hmacKey.ts [hunk 2]
```diff
@@ -339,5 +355,57 @@ export class HmacKey extends ServiceObject<HmacKeyMetadata | undefined> {
...
+    metadata: Metadata,
+    options: SetMetadataOptions,
+    callback: MetadataCallback
+  ): void;
+  setMetadata(
+    metadata: Metadata,
+    optionsOrCallback: SetMetadataOptions | MetadataCallback,
+    cb?: MetadataCallback
+  ): Promise<SetMetadataResponse> | void {
+    // ETag preconditions are not currently supported so retries should be disabled
```
Comment (shaffeeullah) [inline src/hmacKey.ts:387]: unless the retry strategy is `RetryAlways`, right? i made this mistake too in IAM, I will fix it now

File: conformance-test/libraryMethods.ts (hunk 1 above)
Comment (ddelgrosso1) [reply conformance-test/libraryMethods.ts]: It does not.  I spoke with Cathy and verified in testing that only the delete call will utilize instruction count.

Comment (shaffeeullah) [reply conformance-test/libraryMethods.ts]: oh right because the instructions are on the method specified. this makes sense.

**Review by shaffeeullah — APPROVED** [review]
thanks, denis!

**Pull request merged**